include/zephyr/arch/riscv/irq.h

```diff
@@ -29,14 +29,70 @@ extern void arch_irq_enable(unsigned int irq);
 extern void arch_irq_disable(unsigned int irq);
 extern int arch_irq_is_enabled(unsigned int irq);
 
-#if defined(CONFIG_RISCV_HAS_PLIC) || defined(CONFIG_RISCV_HAS_CLIC)
+#if defined(CONFIG_RISCV_HAS_PLIC) || defined(CONFIG_RISCV_HAS_CLIC) || \
+	defined(CONFIG_RISCV_CUSTOM_INTERRUPT_CONTROLLER)
 extern void z_riscv_irq_priority_set(unsigned int irq,
 				     unsigned int prio,
 				     uint32_t flags);
 #else
 #define z_riscv_irq_priority_set(i, p, f) /* Nothing */
 #endif /* CONFIG_RISCV_HAS_PLIC || CONFIG_RISCV_HAS_CLIC */
 
+#if defined(CONFIG_RISCV_CUSTOM_INTERRUPT_CONTROLLER)
+
+/**
+ * @brief Enable an interrupt line for RISC-V custom interrupt controller.
+ *
+ * This routine enables an interrupt line for RISC-V custom interrupt controller.
+ * Should be implemented by custom interrupt controller.
+ *
+ * @param irq IRQ number to enable
+ */
+void z_irq_enable(unsigned int irq);
```

Comment:
Should it be `z_soc_*`? Following the existing [zephyr/arch/arm/irq.h](https://github.com/zephyrproject-rtos/zephyr/blob/133f20abc58bc195fe92e131cbdc1f0b3b566e27/include/zephyr/arch/arm/irq.h#L54-L57).

Reply:
Hi @xingrz , 

We were considering such prefix, but decided that it wouldn't be a good choice, please see:
https://github.com/zephyrproject-rtos/zephyr/pull/63214#issuecomment-1759126209